skrf/network.py

```diff
@@ -5999,13 +5999,12 @@ def innerconnect_s(A: np.ndarray, k: int, l: int) -> np.ndarray:
 
     # create temporary matrices for calculation
     det = (Akl * Alk - Akk * All)
-    tmp_a = (Ael * Alk + Aek * All) / det
-    tmp_b = (Ael * Akk + Aek * Akl) / det
+    tmp_a = Ael * (Alk / det) + Aek * (All / det)
+    tmp_b = Ael * (Akk / det) + Aek * (Akl / det)
```

Comment:
Why is this more optimized than the previous implementation?

Reply:
The shape of ndarrays (`Alk`, `All`, `Akk`, `Akl`, and `det`) are (n_freq,), but the shape of ndarrays (`Ael`, `Aek`) are (n_freq, nA-2).
So I believe that performing `Ael * (Alk / det) + Aek * (All / det)` is faster than `(Ael * Alk + Aek * All) / det`, because it is appropriate to use two small-scale computations to replace the large-scale computation performed via broadcasting.

Reply:
A demo may be able to describe my idea more clearly,
```python
>>> import numpy as np
>>> from timeit import timeit
>>> 
>>> a=np.random.rand(4,101)
>>> b=np.random.rand(4,101)
>>> c=np.random.rand(101)
>>> d=np.random.rand(101)
>>> e=np.random.rand(101)
>>>
>>> timeit(lambda: (a*c)/e) # Perform large scale computation (a * c) / e at first
2.020706125003926
>>> timeit(lambda: a*(c/e)) # Perform small scale computation a * (c / e) at first
1.373241916000552
>>> timeit(lambda: (a*c+b*d)/e)
3.5458894160037744
>>> timeit(lambda: a*(c/e)+b*(d/e))
3.1978541249991395
```

Reply:
Interesting! Are you using some tools to find these improvements?

Reply:
Usually, I use tools like `cProfile` and `VizTracer` to profile my code. However, optimizations such as swapping division orders come from extensive testing and experimentation on my part. 😂